src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -248,7 +248,7 @@ private bool BeforeExecuteCommand(bool speculative, TypeCharCommandArgs args, Co
 
         private static bool CanHaveSemicolon(SyntaxNode currentNode)
         {
-            if (currentNode.IsKind(SyntaxKind.FieldDeclaration, SyntaxKind.DelegateDeclaration, SyntaxKind.ArrowExpressionClause))
```

Comment:
I'm afraid this can move the caret in cases where it's not appropriate. I can't think of such a scenario, but if I'll do it, I'll add another `if` below the existing one:

```csharp
            if (currentNode.IsKind(SyntaxKind.EqualsValueClause) && currentNode.IsParentKind(SyntaxKind.PropertyDeclaration))
            {
                return true;
            }
```

Then, the below change in `TryGetCaretPositionToMove` (line 343) will have to change to `EqualsValueClause` instead of `PropertyDeclaration`.